webhook/workspace/handler/immutable.go

```diff
@@ -177,7 +187,14 @@ func changePermitted(oldObj, newObj runtime.Object) bool {
 		log.Error(fmt.Errorf("Object %s is not a valid k8s object: does not have metadata", newObj.GetObjectKind()), "Failed to compare objects")
 		return false
 	}
-	newMeta.SetFinalizers(oldMeta.GetFinalizers())
-	newMeta.SetDeletionTimestamp(oldMeta.GetDeletionTimestamp())
-	return reflect.DeepEqual(oldMeta, newMeta)
+	oldLabels, newLabels := oldMeta.GetLabels(), newMeta.GetLabels()
+	oldAnnotations, newAnnotations := oldMeta.GetAnnotations(), newMeta.GetAnnotations()
+	if oldLabels[constants.WorkspaceCreatorLabel] != newLabels[constants.WorkspaceCreatorLabel] {
+		return false
+	}
+	if oldAnnotations[constants.WorkspaceRestrictedAccessAnnotation] == "true" &&
+		newAnnotations[constants.WorkspaceRestrictedAccessAnnotation] != "true" {
+		return false
```

Comment:
Since we allow to modify some pieces, it makes sense to provide message here, what exactly user tried to modify but it's forbidden, like, when I tried to update creator, I got `object is owned by workspace and cannot be updated` but it could be something like `creator label is managed by devworkspace operator and can't be updated`.

For restricted access it can be: `restricted access should be configure on DevWorkspace level`

Reply:
Great idea, done.